### DISCUSSION TITLE
Any way that CR CE does not think multivolume files are missing an included volume

### DISCUSSION THREAD

**Discussion by saskir12** [maforget/ComicRackCE#65] [Ideas]
Something which bothered me for quite some time. There are omnibusses out there which include 2-3 volumes of a comic/manga. Now CR thinks in the automatically created "Series Gap" that some volumes are missing.

Now the volume number is accordingly 1-3 for the next one 4-6, etc. An example would be "Real account" which started with single volume but after volume 9 it released with 3 in 1 books. Or "Yamada-kun and the seven witches" which even started later with the omnibusses. Now I could simply name volume 9-11 as 9 and 12-14 as 10. But after a refresh with CR it goes back to the other volume numbering. Splitting is also not the best idea as it will not find the metadate of those inbetween volumes and has the plot of all 3 in one.

So any way to get CR to recognize that there is no missing volume from 9-11 and 12-14?

**Comment by maforget** [comment]
Check d2122d21d9f43aedd27507c7c6e16ddcdddd8a97 and let me know, if that works.

**Reply by saskir12** [reply]
Worked like a charm. One press of "F5" and those series in "Series Gap" were gone. Thanks for the fast solution.

**Reply by saskir12** [reply]
Ok little Update. Don't know if it was because I installed the latest version 30 minutes ago but I am again seeing multivolume issues in the "Series Gap" Smartlist. After I clicked on the "Edit" function of the smartlist it removed those volumes again. Even though I did not change a thing.

Any Idea?

**Reply by maforget** [reply]
Lists are cached in the DB, by Editing you force an update. Just like F5. Also it only works on books where the number is of style "#-#".

**Reply by saskir12** [reply]
Yeah have my books as 1-2, etc. A normal refresh did not work as I tried F5. Even worse the volumes did come back after pressing F5.

**Reply by maforget** [reply]
There are probably many reason why. First I don't know if you are talking about just a Smart List updating or the Gab Information value updating on a static list (Like Series Real account, without any Gap Info.) Also take into account that Volume is also used to determine if it's the same series, so having books where some have Volume and some not will not work to determine the series.

What is probably happening when changing the way these value work is that they are still cached in the database. So even if you press F5 it returns the database info. What I would try is in a fresh config (try adding `-ac` followed by a name, like `-ac Test`). It will give you a fresh environment /database where you can test some things.

I am not certain that a simple F5 is enough to force refresh everything. Regarding Series Statistics, it seems that when Cache is enable it always returns the same value. Although both Cache and not Cache seem to have the same code, so I am not certain. 

For example using a Detail Smart List, showing Gap Information that only contains Real Account:

```
Name "Real Account"
Match Match All
{
    [Series] equals "Real Account",
    [Volume] equals "2016"
}
``` 

I change 9-11 to 9-10, and press F5, the Gap Information arrow appears. I can press it as many time as I want it stays that way. Change 9-10 back to 9-11, and press F5, it disappears.

I've tried with a Gaps Smart List:

```
Name "Gaps"
Match All
{
    [Series: Gaps] is greater "0",
    Match Any
    {
        [Series: Start of Gap] equals yes,
        [Series: End of Gap] equals yes
    }
}
``` 

And changing the value would update the list automatically. I tried by deleting the book from the library and adding it back. Inside Folder view. I haven't seen what you mention.

Keep in mind that the DB is updated when exiting the program, so maybe a solution would be to delete the smart list, exit and recreate it. I would try that to see if it's the list caching.

I would also try these settings, that you can either use as Command Line Switch or by editing the `ComicRack.ini`. (Check them all [here](https://github.com/maforget/ComicRackCE/wiki/Command-Line-Switches).)

`-qcm Disabled` will disable cache mode (or `QueryCacheMode` in `ComicRack.ini`)
`-dnlqc` should not use the cache smart list from the database. (or `DoNotLoadQueryCaches` in `ComicRack.ini`, this one doesn't already exists, you will need to add it)

Please send me exact steps to reproduce what you experienced from a fresh config, so I can check on it.
